tests/10apidoc/30room-create.pl

```diff
@@ -82,6 +81,88 @@
       });
    };
 
+test "POST /createRoom makes a room with a name",
+   requires => [ $user_fixture, local_user_fixture(),
+                 qw( can_create_private_room )],
+
+   proves => [qw( can_createroom_with_name )],
+
+   do => sub {
+      my ( $user ) = @_;
+
+      do_request_json_for( $user,
+         method => "POST",
+         uri    => "/r0/createRoom",
+
+         content => {
+            name => "Test Room"
+         },
+      )->then( sub {
+         my ( $body ) = @_;
+
+         assert_json_keys( $body, qw( room_id ));
+         assert_json_nonempty_string( $body->{room_id} );
+
+         my ( $room_id ) = $body->{room_id};
+
+         do_request_json_for( $user,
```

Comment:
ok so the problem with this is that it will be flaky when run against a synapse with a worker config, because the client_reader may not have got the memo about the new room when we get here.

A more reliable way is to call `matrix_create_room_synced` and see what you get back in the /sync response. #481 will probably be helpful here.